app/scripts/migrations/121.2.test.ts

```diff
@@ -0,0 +1,273 @@
+import { migrate, version } from './121.2';
+
+const oldVersion = 121.1;
+
+describe(`migration #${version}`, () => {
+  it('updates the version metadata', async () => {
+    const oldStorage = {
+      meta: { version: oldVersion },
+      data: {},
+    };
+
+    const newStorage = await migrate(oldStorage);
+
+    expect(newStorage.meta).toStrictEqual({ version });
+  });
+
+  it('Does nothing if `networkConfigurations` or `providerConfig` are not in the network controller state', async () => {
+    const oldState = {
+      NetworkController: {
+        selectedNetworkClientId: 'mainnet',
+      },
+    };
+
+    const transformedState = await migrate({
+      meta: { version: oldVersion },
+      data: oldState,
```

Comment:
Nit: It's possible that this test might pass even when changes were made to state, because the assertion is using the same object reference as we're passing in. So if it were mutated, it still would match.

Our migration copy pasta includes a clone step, so in practice this is guaranteed not to happen, but if we mistakenly remove that clone step the test might pass when it should fail. To guard against this, I'd recommend using `cloneDeep` to clone the state inputs in each of these tests.

Reply:
https://github.com/MetaMask/metamask-extension/pull/26671/commits/e8491d8b23f9f2ed08f48a160b197c161a411cc1